pkg/domain/entities/images.go

```diff
@@ -165,6 +166,10 @@ type ImagePullReport struct {
 	Images []string `json:"images,omitempty"`
 	// ID contains image id (retained for backwards compatibility)
 	ID string `json:"id,omitempty"`
+	// Progress contains detailed information about download progress
+	Progress *jsonmessage.JSONProgress `json:"progressDetail,omitempty"`
+	// Status contains a short progress description
+	Status string `json:"status,omitempty"`
```

Comment:
In it's current form, I consider this to be a breaking change.  Previously, all data was part of `Stream`.  Now the data is scattered across `Stream` and `Status`.  If a `podman-remote` prior to this change pulls from the new service the output looks as follows:
```
^Cpodman (main) $ ./bin/podman-remote rmi -af && ./bin/podman-remote pull nginx                                                                 
Resolved "nginx" as an alias (/home/vrothberg/.cache/containers/short-name-aliases.conf)                                                        
Trying to pull docker.io/library/nginx:latest...                                                                                                
Getting image source signatures                                                                                                                 
Copying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blobCopying blob
Copying blobCopying blobCopying configCopying configWriting manifest to image destination                                                       
Storing signatures                                                                                                                              
41b0e86104ba681811bf60b4d6970ed24dd59e282b36c352b8a55823bbb5e14a                                                                                
```

That means, unless we find a backwards compatible way, rendering the progress must be opt-in.  Older clients must continue working with newer servers (unless the major versions differ).